resources/suite-runner.mjs

```diff
@@ -89,13 +112,108 @@ export class SuiteRunner {
         this.#suiteResults.tests[test.name] = { tests: { Sync: syncTime, Async: asyncTime }, total: total };
         this.#suiteResults.total += total;
 
-        if (this.#client?.didRunTest)
-            await this.#client.didRunTest(this.#suite, test);
+        await this._updateClient(test);
     };
+
+    async _updateClient(test, suite = this.#suite) {
+        if (this.#client?.didRunTest)
+            await this.#client.didRunTest(suite, test);
+    }
 }
 
-// FIXME: implement remote steps
-class RemoteSuiteRunner extends SuiteRunner {}
+export class RemoteSuiteRunner extends SuiteRunner {
+    #appId;
+
+    get appId() {
+        return this.#appId;
+    }
+
+    set appId(id) {
+        this.#appId = id;
+    }
```

Comment:
this can definitely be just a public field if we have public getters *and* setters.